x/fees/keeper/integration_test.go

```diff
@@ -0,0 +1,621 @@
+package keeper_test
+
+import (
+	"math"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/tharsis/ethermint/crypto/ethsecp256k1"
+	"github.com/tharsis/ethermint/encoding"
+	"github.com/tharsis/ethermint/tests"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+	"github.com/tharsis/evmos/v3/app"
+	"github.com/tharsis/evmos/v3/testutil"
+	"github.com/tharsis/evmos/v3/x/fees/types"
+	inflationtypes "github.com/tharsis/evmos/v3/x/inflation/types"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+	claimstypes "github.com/tharsis/evmos/v3/x/claims/types"
+)
+
+var contractCode = "600661000e60003960066000f300612222600055"
+
+// Uses CREATE opcode to deploy the above contract and emits
+// log1(0, 0, contractAddress)
+var factoryCode = "603061000e60003960306000f3007f600661000e60003960066000f300612222600055000000000000000000000000600052601460006000f060006000a1"
+
+var _ = Describe("While", Ordered, func() {
+	feeCollectorAddr := s.app.AccountKeeper.GetModuleAddress(authtypes.FeeCollectorName)
+	claimsDenom := claimstypes.DefaultClaimsDenom
+	evmDenom := evmtypes.DefaultEVMDenom
+	accountCount := 4
+
+	// account initial balances
+	initAmount := sdk.NewInt(int64(math.Pow10(18) * 2))
+	initBalance := sdk.NewCoins(
+		sdk.NewCoin(claimsDenom, initAmount),
+		sdk.NewCoin(evmDenom, initAmount),
+	)
+	totalAmount := sdk.NewCoin(claimsDenom, initAmount.MulRaw(int64(accountCount)))
+
+	var (
+		deployerKey      *ethsecp256k1.PrivKey
+		userKey          *ethsecp256k1.PrivKey
+		deployerAddress  sdk.AccAddress
+		userAddress      sdk.AccAddress
+		params           types.Params
+		contractAddress  common.Address
+		contractAddress1 common.Address
+		contractAddress2 common.Address
+		factoryAddress   common.Address
+		factoryNonce     uint64
+	)
+
+	BeforeAll(func() {
+		s.SetupTest()
+
+		params = s.app.FeesKeeper.GetParams(s.ctx)
+		params.EnableFees = true
+		s.app.FeesKeeper.SetParams(s.ctx, params)
+
+		// mint coins for claiming and send them to the claims module
+		coins := sdk.NewCoins(totalAmount)
+		err := testutil.FundModuleAccount(s.app.BankKeeper, s.ctx, inflationtypes.ModuleName, coins)
+		s.Require().NoError(err)
+
+		// setup deployer account
+		deployerKey, _ = ethsecp256k1.GenerateKey()
+		deployerAddress = getAddr(deployerKey)
```

Comment:
```suggestion
		deployerKey, deployerAddr = ethsecp256k1.GenerateKey()
		deployerAddress = sdk.AccAddresS(deployerAddr.Bytes())
```

Reply:
The function signature is `func ethsecp256k1.GenerateKey() (*ethsecp256k1.PrivKey, error)`. Did you meant another function?

Reply:
yeah there's another function to generate the address in the ethermint tests package